lib/systemFlake.nix

```diff
@@ -94,8 +101,16 @@ let
   getChannels = system: self.pkgs.${system};
   getNixpkgs = host: (getChannels host.system).${host.channelName};
 
-  configurationBuilder = hostname: host': (
+  configurationBuilder = reverseDnsFqdn: host': (
     let
+      dnsLabels = reverseList (partitionString "\\." reverseDnsFqdn);
+      hostname = builtins.head dnsLabels;
+      domain = let
+        domainLabels = builtins.tail dnsLabels;
+      in
+        if domainLabels == [] then (lib.mkDefault null) # null is the networking.domain option's default
+        else builtins.concatStringsSep "." domainLabels;
```

Comment:
Would be nice to move these builtins to the top like we do with the rest of them

Reply:
Addressed in https://github.com/gytis-ivaskevicius/flake-utils-plus/pull/69/commits/0d915599218f910c63ced0c3d0c1732f72b98c30